mass_mailing_switzerland/models/recurring_contract.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Nathan Fluckiger <nathan.fluckiger@hotmail.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import models, fields, api
+
+
+class RecurringContract(models.Model):
+
+    _inherit = 'recurring.contract'
+
+    campaign_id = fields.Many2one('utm.campaign')
+
+    @api.model
+    def create(self, vals):
+        origin = self.env['recurring.contract.origin'].search([('id', '=', vals['origin_id'])])
```

Comment:
This will fail if `origin_id` is not set at creation. You should first test `if 'origin_id' in vals:`